aip/general/0236.md

```diff
@@ -0,0 +1,321 @@
+  previews of different fields-under-test.
+
+#### list
+
+- List **must** support filtering on the state of each experiment.
+  - Filtering on `PreviewMetadata` indicates which experiments are actively 
+    previewed.
+
+### Live policy API changes
+
+The following guidance **must** be followed for live resources which use
+experiments as a nested collection for preview.
+
+#### delete
+
+- A delete of the live policy **must** delete all experiments.
+- To maintain the experiments while negating the effect of the live policy, the
+  live policy **must** be changed to a no-op policy instead of using delete.
+
+### Custom method APIs
+
+#### Preview
+
+```proto
+// Starts the preview of a PolicyExperiment. This starts the log generation to
+// evaluate the PolicyExperiment.
+rpc PreviewPolicyExperiment(PreviewPolicyExperimentRequest) returns (google.longrunning.Operation) {
+  option (google.api.http) = {
+    post: "/v1/{name=policies/*/experiments/*}:preview"
+    body: "*"
+  };
+  option (google.longrunning.operation_info) = {
+    response_type: "PolicyExperiment"
+    metadata_type: "PreviewPolicyExperimentMetadata"
+  };
+}
+
+message PreviewPolicyExperimentRequest {
+  // The name of the PolicyExperiment.
+  string name = 1;
+}
+
+message PreviewPolicyExperimentMetadata {}
+```
+- This custom method is required.
+- This method **must** trigger the system to start generating logs for preview.
+- When the method is called, the system **must** set values for the
+  `log_prefix`, and `start_time` in `PreviewMetadata` and set the state to
+  `ACTIVE`.
+- If the method is called on an experiment with an empty set of rules, the
+  system **must** simulate the deletion of the live policy.
+
+#### Stop
+
+```proto
+// Stops the preview of a PolicyExperiment. This stops the log generation to
+// evaluate the PolicyExperiment.
+rpc StopPolicyExperiment(StopPolicyExperimentRequest) returns (google.longrunning.Operation) {
+  option (google.api.http) = {
+    post: "/v1/{name=policies/*/experiments/*}:stop"
+    body: "*"
+  };
+  option (google.longrunning.operation_info) = {
+    response_type: "PolicyExperiment"
+    metadata_type: "StopPolicyExperimentMetadata"
+  };
+}
+
+message StopPolicyExperimentRequest {
+  // The name of the PolicyExperiment.
+  string name = 1;
+}
+
+message StopPolicyExperimentMetadata {}
+```
+- This custom method is required.
+- This method **must** trigger the system to stop generating logs for preview.
+- When the method is called, the `state` **must** be set to `INACTIVE` and
+  `stop_time` **must** be set to the current time.
+
+#### Commit
+
+The resource may expose a new API called Commit to promote the experimental
+configuration of a resource to the live version and perform necessary cleanup.
+When used, the system copies the fields that are under test from the experiment
+into the live policy and then deletes the experiment.
+
+Declarative clients **may** manually copy fields from an experiment into the
```

Comment:
How are we expecting declarative clients to start and stop the experiments? (My understanding was that declarative clients really don't get on well with custom methods. I may have misunderstood.)